### DISCUSSION TITLE
Linear advance on Ender 3 V2

### DISCUSSION THREAD

**Discussion by RoP09** [mriscoc/Ender3V2S1#300] [Show and tell]
Big thumbs up for that one!

Using K = 0.065 gives fantastic results I could never reach by adjusting the retraction settings and all kinds of slice tricks so far! No more blobs or gaps at retraction points.
Printing single wall RC plane parts now looks 10 times better thanks to this firmware here 'Experimental-Ender3V2-422-BLTUBL-LA-20220817.bin'

Using it on an Ender 3 V2 with Stock 4.2.2 mainboard and a Micro Swiss direct drive extruder (+ Creality Spider hotend), seems I already have the 2209 drivers then?

**Comment by mriscoc** [comment]
Recently updates in Marlin's Linear Advance feature allowed us to use it with the TMC2208 drivers in stantalone mode inclusive.

**Reply by RoP09** [reply]
What about adding a switch for the GUI to enable it? Or keeping it active all the time with K=0 (will not harm without a M900 command as i understood)?

**Reply by mriscoc** [reply]
K = 0 disables linear advance.

**Comment by RoP09** [comment]
I mean why not adding it to the "normal" release versions instead of making a special version? If K=0 is default it should not harm, right? (for users that cannot use it)

**Reply by mriscoc** [reply]
Because it is an experimental feature, and because it takes program memory space, the RCT6 devices are close to the limit. Another thing, currently "Linear Advance" configuration disables "Classic Jerk" and "S Curve Acceleration".

**Reply by MetalFranke** [reply]
sad but true. I tried to build a "Ender3S1-F4-LA" configuration but it does not fit into 256k flash (288 bytes to large).
Do you see any possibility to use LA with the F4 controller?

**Reply by mriscoc** [reply]
Check in the Configuration.h file after the define of DWIN_LCD_PROUI there are some features that you can disable according to your preferences.

https://github.com/mriscoc/Ender3V2S1/blob/cda931060f3121fab2ed56382a3bfaf5837d7fab/Marlin/Configuration.h#L3152-L3161

**Reply by Harvie** [reply]
> currently "Linear Advance" configuration disables "Classic Jerk" and "S Curve Acceleration".

My understanding is that LA is more important to print quality than Jerk or S-Curves (at least on bowden based machines) right?

So what acceleration algorithm would you reccomend to use with LA instead of these?

At least in the meantime before https://github.com/MarlinFirmware/Marlin/issues/22547 gets fixed...

**Comment by Harvie** [comment]
If you're interrested. I've found more info about how Marlin did fixed Linear Advance at TMC2208:

https://github.com/MarlinFirmware/Marlin/issues/16076
https://github.com/MarlinFirmware/Marlin/pull/24533
